src/sdam/server.ts

```diff
@@ -310,14 +310,16 @@ export class Server extends TypedEventEmitter<ServerEvents> {
     this.incrementOperationCount();
     if (conn == null) {
       try {
-        conn = await this.pool.checkOut();
+        conn = await this.pool.checkOut({ timeout: options.serverSelectionTimeout });
```

Comment:
let's flag this path to only occur when csot is on explicitly, same logic just a bit different looking code